### PR TITLE
Update reference assembly project.json files to use -rc2-23602 build

### PR DESCRIPTION
**Pull request by iamjasonp**
Updated the ref/ project.json references as well to be consistent with what we reference elsewhere.

### CODE REVIEW

File: src/System.ServiceModel.NetTcp/ref/project.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
     "System.ObjectModel": "4.0.0",
     "System.Runtime": "4.0.0",
     "System.Net.Primitives": "4.0.10",
-    "System.ServiceModel.Primitives": "4.1.0-beta-*",
+    "System.ServiceModel.Primitives": "4.1.0-rc2-23602",
```
Comment (zhenlan) [inline src/System.ServiceModel.NetTcp/ref/project.json]: Doesn't 4.1.0-rc2-* work?

Comment (iamjasonp) [inline src/System.ServiceModel.NetTcp/ref/project.json]: Very shortly project.lock.json files will go away and we have been advised by weshaggard to reference specific versions in our project.json files.

Comment (iamjasonp) [inline src/System.ServiceModel.NetTcp/ref/project.json]: See #565

Comment (zhenlan) [inline src/System.ServiceModel.NetTcp/ref/project.json]: I see. Thanks.

Comment (roncain) [inline src/System.ServiceModel.NetTcp/ref/project.json]: Does this build in TFS?  I'm having TFS build issues right now and can't verify.

**Comment by roncain** [comment]
LGTM, subject to building in TFS.

**Comment by iamjasonp** [comment]
I ran into issues building into TFS, but seemed unrelated to this change. Unfortunately gated build in TFS is also broken, so I can't even validate using that.

File: src/System.ServiceModel.NetTcp/ref/project.json (hunk 1 above)
Comment (weshaggard) [inline src/System.ServiceModel.NetTcp/ref/project.json]: Yes please no more * versions in the project.json files.

**Comment by roncain** [comment]
BTW, I'm trying to move forward to 23608 to see it if allows Linux OuterLoops to pass, but you should probably merge this one anyway to get off the beta dependency.

**Comment by iamjasonp** [comment]
I've waited a few days for this moment, but I see a window where the internal TFS build is clean. I'm submitting this as a gated checkin to ensure successful TFS building. Wish me luck.

**Comment by iamjasonp** [comment]
*sigh. TFS busted again.

**Comment by roncain** [comment]
@iamjasonp -- would you mind waiting on this PR until I get my version bump PR merged?  I'll incorporate your changes and do it all in one PR.  Linux tests don't work unless we go up to 23608.
Please leave this PR open for now in case this plan changes.

**Comment by roncain** [comment]
@iamjasonp -- okay, I believe PR #600 now incorporates the edits you made in this PR plus increases the package dependency version to the latest one so that we can run OuterLoop in Linux again.

Could I ask you to just confirm the project.json edits you made in this PR were properly made in #600 too?  Once we're good with that, I would like to merge #600 and **close** this current PR without merging.

**Comment by iamjasonp** [comment]
I'm good with closing the PR when #600 is merged. I checked #600 and confirmed that the changes made in this PR have also been made in that PR.

**Comment by iamjasonp** [comment]
Closing as #600 is now merged.

**Pull request closed without merging**